NullPointerException in GradleBuildInfoExtractor when module dependencies are null

**Describe the bug**
When running the artifactoryDeploy task for a single module in a multi-module project, it fails with a NullPointerException, since the module dependencies are null.

Stacktrace:
`Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':artifactoryDeploy'.
Caused by: java.lang.NullPointerException
	at org.jfrog.gradle.plugin.artifactory.extractor.GradleBuildInfoExtractor.lambda$extract$1(GradleBuildInfoExtractor.java:109)
	at org.jfrog.gradle.plugin.artifactory.extractor.GradleBuildInfoExtractor.extract(GradleBuildInfoExtractor.java:106)
	at org.jfrog.gradle.plugin.artifactory.task.DeployTask.prepareAndDeploy(DeployTask.java:114)
	at org.jfrog.gradle.plugin.artifactory.task.DeployTask.collectProjectBuildInfo(DeployTask.java:55)
	at org.jfrog.gradle.plugin.artifactory.task.DeployTask.taskAction(DeployTask.java:49)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:49)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:727)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:694)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:568)
	at `

**Versions**
Gradle Build Info Extractor 4.17.2
Gradle 6.0.1
Android Gradle Plugin 3.5.3

Question:
1. In which scenario would module dependencies be set to null instead of being just an empty list that is expected?

--- a/build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/GradleBuildInfoExtractor.java
+++ b/build-info-extractor-gradle/src/main/groovy/org/jfrog/gradle/plugin/artifactory/extractor/GradleBuildInfoExtractor.java
@@ -20,18 +20,7 @@ import org.apache.commons.lang.StringUtils;
 import org.gradle.api.Project;
 import org.gradle.api.logging.Logger;
 import org.gradle.api.logging.Logging;
-import org.jfrog.build.api.Agent;
-import org.jfrog.build.api.BlackDuckProperties;
-import org.jfrog.build.api.Build;
-import org.jfrog.build.api.BuildAgent;
-import org.jfrog.build.api.Governance;
-import org.jfrog.build.api.Issue;
-import org.jfrog.build.api.IssueTracker;
-import org.jfrog.build.api.Issues;
-import org.jfrog.build.api.LicenseControl;
-import org.jfrog.build.api.MatrixParameter;
-import org.jfrog.build.api.Vcs;
-import org.jfrog.build.api.Module;
+import org.jfrog.build.api.*;
 import org.jfrog.build.api.builder.BuildInfoBuilder;
 import org.jfrog.build.api.builder.PromotionStatusBuilder;
 import org.jfrog.build.api.release.Promotion;
@@ -99,14 +88,16 @@ public class GradleBuildInfoExtractor implements BuildInfoExtractor<Project> {
         bib.durationMillis(durationMillis);
 
         Set<File> moduleFilesWithModules = moduleInfoFileProducers.stream()
-            .filter(ModuleInfoFileProducer::hasModules)
-            .flatMap(moduleInfoFileProducer -> moduleInfoFileProducer.getModuleInfoFiles().getFiles().stream())
-            .collect(Collectors.toSet());
+                .filter(ModuleInfoFileProducer::hasModules)
+                .flatMap(moduleInfoFileProducer -> moduleInfoFileProducer.getModuleInfoFiles().getFiles().stream())
+                .collect(Collectors.toSet());
 
         moduleFilesWithModules.forEach(moduleFile -> {
             try {
                 Module module = ModuleExtractorUtils.readModuleFromFile(moduleFile);
-                if (!module.getArtifacts().isEmpty() || !module.getDependencies().isEmpty()) {
+                List<Artifact> artifacts = module.getArtifacts();
+                List<Dependency> dependencies = module.getDependencies();
+                if ((artifacts != null && !artifacts.isEmpty()) || (dependencies != null && !dependencies.isEmpty())) {
                     bib.addModule(module);
                 }
             } catch (IOException e) {
@@ -132,7 +123,7 @@ public class GradleBuildInfoExtractor implements BuildInfoExtractor<Project> {
         bib.artifactoryPrincipal(artifactoryPrincipal);
 
         String artifactoryPluginVersion = clientConf.info.getArtifactoryPluginVersion();
-        if (StringUtils.isBlank(artifactoryPluginVersion)){
+        if (StringUtils.isBlank(artifactoryPluginVersion)) {
             artifactoryPluginVersion = "Unknown";
         }
         bib.artifactoryPluginVersion(artifactoryPluginVersion);

